addons/web/static/tests/core/action_swiper.test.js

```diff
@@ -0,0 +1,717 @@
+        static components = { ActionSwiper };
+        static template = xml`
+                    <div class="d-flex">
+                        <ActionSwiper
+                            onRightSwipe = "{
+                                action: () => this.onRightSwipe(),
+                                icon: 'fa-circle',
+                                bgColor: 'bg-warning'
+                            }"
+                            onLeftSwipe = "{
+                                action: () => this.onLeftSwipe(),
+                                icon: 'fa-check',
+                                bgColor: 'bg-success'
+                            }">
+                                <div class="target-component" style="width: 250px; height: 80px">Swipe in both directions</div>
+                        </ActionSwiper>
+                    </div>
+                `;
+        onRightSwipe() {
+            expect.step("onRightSwipe");
+        }
+        onLeftSwipe() {
+            expect.step("onLeftSwipe");
+        }
+    }
+    await mountWithCleanup(Parent);
+    const swiper = queryFirst(".o_actionswiper");
+    const targetContainer = queryFirst(".o_actionswiper_target_container");
+    const dragHelper = await contains(swiper).drag({
+        position: {
+            clientX: 0,
+            clientY: 0,
+        },
+    });
+    await dragHelper.moveTo(swiper, {
+        position: {
+            clientX: (3 * swiper.clientWidth) / 4,
+            clientY: 0,
+        },
+    });
+    expect(targetContainer.style.transform).toInclude("translateX", {
+        message: "target has translateX",
+    });
+    // Touch ends before the half of the distance has been reached to the left
+    await dragHelper.moveTo(swiper, {
+        position: {
+            clientX: -swiper.clientWidth / 2 + 1,
+            clientY: 0,
+        },
+    });
+
+    await dragHelper.drop();
+
+    expect(targetContainer.style.transform).not.toInclude("translateX", {
+        message: "target does not have a translate value",
+    });
+
+    // Touch ends once the half of the distance has been crossed to the left
+    await swipeLeft(
+        ".o_actionswiper",
+        {
+            position: {
+                clientX: swiper.clientWidth / 2,
+                clientY: 0,
+            },
+        },
+        {
+            position: {
+                clientX: -swiper.clientWidth - 1,
+                clientY: 0,
+            },
+        }
+    );
+    await advanceTime(500);
+    expect.verifySteps(["onLeftSwipe"]);
+    // Touch ends once the half of the distance has been crossed to the right
+    await swipeRight(
+        ".o_actionswiper",
+        {
+            position: {
+                clientX: swiper.clientWidth / 2,
+                clientY: 0,
+            },
+        },
+        {
+            position: {
+                clientX: swiper.clientWidth + 1,
+                clientY: 0,
+            },
+        }
+    );
```

Comment:
```suggestion
    await swipeRight(".o_actionswiper");
```

Position not needed test valid/green without it.